ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/ConnectionImple.java

```diff
@@ -1002,7 +1003,7 @@ protected final synchronized void registerDatabase() throws SQLException
 							jdbcLogger.i18NLogger.debug_closingconnection(_theConnection.toString());
 
 							delayClose = true;
-							tx.registerSynchronization(new ConnectionSynchronization(this, needsClose));
+							jtaPropertyManager.getJTAEnvironmentBean().getTransactionSynchronizationRegistry().registerInterposedSynchronization(new ConnectionSynchronization(this, needsClose));
```

Comment:
So this is the only functional change made by this PR. Please can you explain how it fixes the two linked JIRAS.